app/src/actions/tasks.ts

```diff
@@ -31,15 +33,24 @@ export function loadTasks() {
 export function addTask(task: Task, onComplete: Function) {
   return (dispatch) => {
     delete task._id;
-    return Tasks.add(task)
-      .subscribe((res) => {
-        if (res.length === 1) {
-          dispatch({
-            type: TASK_ADDED,
-            payload: res[0]
-          });
-        }
-      }, null, onComplete); }
+
+    const taskAdd$ = Tasks.add(task).publish();
+    taskAdd$.connect();
+
+    taskAdd$.subscribe(res=> {
+      if (res.length === 1) {
+        dispatch({
+          type: TASK_ADDED,
+          payload: res[0]
+        });
+      }
+    })
+
+
+    return taskAdd$;
+
+
+  }
```

Comment:
> The more interesting change (in my opinion) would be having the reducer subscribe to the task observables 

With Redux - the state/actions are based into the reducers by an outside thing. WIth more traditional flux, you do have a 'store listens to x' and sort of reaches outside of itself. With Redux - things are passed in. So having your reducer explicitly be subscribed to a stream is sort of going against that.